packages/elements/src/photon-medication-search/photon-medication-search-component.tsx

```diff
@@ -0,0 +1,350 @@
+    fetchPolicy: 'no-cache'
+  });
+};
+
+async function loadTreatmentOptions(
+  client: ApolloClient<any>,
+  searchText: string
+): Promise<TreatmentOption[]> {
+  const req = await searchTreatmentoptions(client, searchText);
+  return req?.data?.treatmentOptions ?? [];
+}
+
+function getFilteredData(
+  props: ComponentProps,
+  searchText: string,
+  treatmentOptions: TreatmentOption[]
+): (Treatment | PrescriptionTemplate | TreatmentOption)[] {
+  const { store } = CatalogStore;
+
+  if (store.catalogs.data.length === 0) return [];
+
+  const catalogData = [
+    ...(props.offCatalogOption ? [props.offCatalogOption as Treatment] : []),
+    ...store.catalogs.data[0].templates.map((x) => x as PrescriptionTemplate),
+    ...store.catalogs.data[0].treatments.map((x) => x as Treatment),
+    ...treatmentOptions
+  ];
+
+  const searchTextLowerCase = searchText.toLowerCase();
+  if (searchTextLowerCase.length === 0) return catalogData;
+
+  return catalogData.filter((x) => {
+    if ('treatment' in x) {
+      return (
+        x.treatment.name.toLowerCase().includes(searchTextLowerCase) ||
+        x.name?.toLowerCase().includes(searchTextLowerCase)
+      );
+    }
+    return x.name.toLowerCase().includes(searchTextLowerCase);
+  });
+}
+
+export const boldSubstring = (inputString: string, substring: any) => {
+  const substrings = substring.split(' ').filter((part: string) => part.length > 0);
+  const regex = new RegExp(`(${substrings.join('|')})`, 'gi');
+  const parts = inputString.split(regex);
+  return parts.map((part) => {
+    if (substrings.some((sub: string) => sub.toLowerCase() === part?.toLowerCase())) {
+      return <strong class="font-extrabold">{part}</strong>;
+    } else {
+      return part;
+    }
+  });
+};
+
+function displayTreatment(
+  t: Treatment | TreatmentOption,
+  groupAccess: boolean,
+  searchText: string
+) {
+  return groupAccess ? (
+    <p class="text-sm whitespace-normal leading-snug my-1">{boldSubstring(t.name, searchText)}</p>
+  ) : (
+    t.name || ''
+  );
+}
+
+function displayPrescriptionTemplate(
+  t: PrescriptionTemplate,
+  groupAccess: boolean,
+  searchText: string
+) {
+  if (groupAccess) {
+    const refills = t.fillsAllowed ? t.fillsAllowed - 1 : 0;
+
+    return (
+      <div class="my-1">
+        <div class="text-sm whitespace-normal font-medium mb-1">
+          {t.name ? <span class="text-blue-600">({boldSubstring(t.name, searchText)}): </span> : ''}
+          {boldSubstring(t.treatment.name, searchText)}
+        </div>
+        <div class="text-xs text-gray-500 truncate">
+          {t.dispenseQuantity} {t.dispenseUnit}, {t.daysSupply}{' '}
+          {t.daysSupply === 1 ? 'day' : 'days'} supply, {refills}{' '}
+          {refills === 1 ? 'refill' : 'refills'}, {t.instructions}
+        </div>
+      </div>
+    );
+  }
+
+  return t.treatment.name;
+}
+
+function getGroupsConfig(props: ComponentProps) {
+  return [
+    {
+      label: 'OFF CATALOG',
+      filter: (t: any) =>
```

Comment:
I would make these `t: { id: string } | undefined`

Reply:
also needed isPrivate and __typename